### PR TITLE
Retrieving K nearest neighbors using faiss 

### PR DESCRIPTION
**Pull request by btseytlin**
The gist of it:
```
inference_model = InferenceModel(trunk=model)
train_vectors = [self.dataset[i][0] for i in range(len(self.dataset))]
inference_model.train_indexer(train_vectors, self.emb_dim)

indices, distances = inference_model.get_nearest_neighbors([train_vectors[0]], k=10)
```

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Looks good, I added it to v0.9.89.dev1

**Comment by AngelaMay27** [comment]
#64

**Comment by gaussiangit** [comment]
@KevinMusgrave Great work! How can I use it to infer on image folder to cluster similar images ?

**Comment by KevinMusgrave** [comment]
@gaussiangit This library is mainly for training and evaluating accuracy. The ```InferenceModel``` just retrieves nearest neighbors given a query image. What you're talking about sounds more like [clustering](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.cluster) or [k-nearest-neighbors classification](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html), so I would look for a library that specifically does those things.
